File: Data Structures & Algorithms/special-array-i/submission-0.java
class Solution {
    public boolean isArraySpecial(int[] nums) {
        if(nums.length == 1) return true;
        boolean odd= false;
        boolean even = false;;
        for(int i=0; i<nums.length-1; i++)
        {
            odd = false;
            even = false;
            if(nums[i]%2==0 && nums[i+1]%2==1 || nums[i]%2==1 && nums[i+1]%2==0)
            {
                odd= true;
                even = true;
            }else return false;
        }
        return (odd==true && even == true);
    }
}

File: Data Structures & Algorithms/search-insert-position/submission-0.cpp
class Solution {
public:
    int binsearch(vector<int>&nums, int low, int high, int tgt)
    {
        int ans=nums.size();
        while(low<=high)
        {
            int mid = (low+high)/2;
            if(nums[mid]==tgt) return mid;
            else if(nums[mid]>tgt)
            {
                ans = mid;
                high=mid-1;
            }else{
                low=mid+1;
            }
        }
        return ans;
    }
    int searchInsert(vector<int>& nums, int target) {
        int n = nums.size();
        int low = 0, high = n-1;
        return binsearch(nums,low,high,target);
    }
};